src/application/node/application.ts

```diff
@@ -23,26 +23,54 @@ export interface BackendApplicationContribution {
 export class BackendApplication {
 
     private app: express.Application;
+    private fs = require('fs');
 
     constructor(
         @inject(ContributionProvider) @named(BackendApplicationContribution)
         protected readonly contributionsProvider: ContributionProvider<BackendApplicationContribution>,
         @inject(ILogger) protected readonly logger: ILogger
     ) { }
 
-    start(port: number = 3000): Promise<void> {
+    createFile(portNumber: number) {
+
+        this.fs.writeFile('port', portNumber.toString(), function (err: string) {
+            if (err) {
+                return console.error(err);
+            }
+        });
+    }
+
+    start(portNumber: number): Promise<void> {
```

Comment:
if the port is undefined then it should be zero in that case node net should assign random port